Topics/Development_Process/Duckdns.md

```diff
@@ -0,0 +1,53 @@
+
+## How do I use DuckDNS?
+
+Using DuckDNS is very straightforward. You can sign up by logging into your twitter, github, or google account. You are then taken to a homepage that shows your account token and the domain name mappings you created. Simply create a new domain name for your application. You do not need to set your application's IP address on this page (the script that will be run does that for us).
+
+![DuckDNS Account Page Image](https://wiki.odroid.com/_media/odroid-xu4/software/nas_guide/170828_ddns01_mosaic.png)
+
+In order to keep DuckDNS up to date, you then need to have your server run a script to tell DuckDNS what its external IP address is. The simplest way to do this (if your server is running on linux) is through the use of a cron job. To do so, on your server:
+
+- Create a folder anywhere called `duckdns`. 
+
+- Navigate into the `duckdns` folder and create a file called `duck.sh`.
+
+- Paste the following into the `duck.sh` file substituting values where appropriate:
+
+```
+echo url="https://www.duckdns.org/update?domains=<YOUR COOL DOMAIN NAME>&token=<YOUR ACCOUNT TOKEN>&ip=" | curl -k -o <PATH TO FOLDER>/duckdns/duck.log -K -
+```
+
+- Make the `duck.sh` file runnable by running `chmod 700 duck.sh`.
+
+- Next, add a job to the running cron process by running `crontab -e` and pasting `*/5 * * * * <PATH TO FOLDER>/duckdns/duck.sh >/dev/null 2>&1` at the bottom of the crontab. This makes the `duck.sh` script run every 5 minutes. If you wish to modify how often this script gets run [this guide](https://www.hostinger.com/tutorials/cron-job) explains the crontab syntax to do so.
+
+- After saving the file, we can finally test that the script works by running `./duck.sh`. This should create a file called `duck.log` which contains the result. If by running `cat duck.log` you get `OK` then you are all good to go! If you get `KO` as output, ensure you copied everything correctly.
+
+If you have an issue getting the cron job to work, the node package [pm2](https://github.com/Unitech/pm2) is helpful in ensuring the `duck.sh` script always runs at boot and every 5 minutes thereafter (or however often you desire). For more details on these instructions and other methods of running the update script on various other devices, visit the DuckDNS [installation page](https://www.duckdns.org/install.jsp).
+
+## A Note for Home Servers
+
+Since DuckDNS updates your _external_ IP address, this will usually refer to your home router. This will be different from the actual IP address of your device on your home network that is running the server. After setting up DuckDNS, in order to tell your router to forward data to where your actual server is running you will need to do port forwarding. More information about what and why you would do this and how to setup your router for port forwarding can be found [here](https://portforward.com/)
```

Comment:
Sorry getting picky but since 'setup' is used as a verb it should be 'set up' https://grammarist.com/spelling/set-up-vs-setup/#:~:text=Setup%20is%20one%20word%20when,set%20up%20the%20computer%E2%80%9D).